### ISSUE TITLE
getTemperture slows down dramatically when no sensor is connected

### ISSUE THREAD

**Issue by matthiaseibl** [Schm1tz1/arduino-tsic#1]
I've noticed that the function getTemperture slows down dramtically if no sensor is connected. I think this is an issue because the timeout in the function readSensor is set too high.

My explanation for the slowing would be the following (please correct me if I'm wrong ;) )
The function reads out the PIN where a TSIC should be to measure the strobe time. Therefore, it waits for the start bit, i.e. until the level goes from HIGH to LOW ("while (TSIC_HIGH)....") and then measures the strobe time as the time when the PIN is in LOW state ("while (TSIC_LOW)..."). If no sensor is connected, the level is (IMO) either LOW or HIGH and does not change its state. Which means that it will run in the timeout either in the first while loop or in the second.

Currently, the timeout is checked with "#define Cancel() if (timeout==0){return 0;}". As timeout is of uint16_t type, it will be 0 again after adding 2^16=65k times 1.

Long story short, a "#define Cancel() if (**timeout>10000**){return 0;}" made my program faster.

What do you think of this change in the code?

btw: here is also some code for arduino&tsic:
http://www.andeanelectronic.com/?Supporte___Arduino_con_TSic306%2CTSic506_y_TSic716

**Comment by Schm1tz1** [comment]
Very nice, well done ! That timeout is a very good point. I didn't have any problems with this as I never worked without sensor connected or having a pin in an undefined state. Just make a pull-request if you are done with your changes, this looks quite useful !

**Comment by matthiaseibl** [comment]
I am very new to github. I hope it worked! Nice work of you, too ;)  This
lib was very helpful for me!

2015-04-01 14:23 GMT+02:00 Roman Schmitz notifications@github.com:
